server/rest/src/main/java/org/infinispan/rest/InfinispanRequestCreator.java

```diff
@@ -12,27 +18,56 @@
  */
 class InfinispanRequestCreator {
 
+   private static final String SEARCH_ACTION = "search";
+   private static final String ACTION_PARAMETER = "action";
+
    /**
     * Creates the appropriate {@link InfinispanRequest} instance based on the raw incoming request.
     */
    static InfinispanRequest createRequest(FullHttpRequest request, ChannelHandlerContext ctx) {
       QueryStringDecoder queryStringDecoder = new QueryStringDecoder(request.uri());
+      Map<String, List<String>> parameters = queryStringDecoder.parameters();
+
+      String[] components = queryStringDecoder.path().substring(1).split("/");
```

Comment:
A comment what we are parsing here would be nice.

Reply:
ok